canonical/lxd-ui
# Extend usePagination hook, move pageSize state into it [WD-2713]

## Done

- Created an adapter for react-components' `usePagination` hook that holds the `pageSize` and exposes it together with a function to update it.
- Reset the current page to 1 when the page size is updated.

Fixes [WD-2713](https://warthogs.atlassian.net/browse/WD-2713)

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described in the [Readme](https://github.com/canonical/lxd-ui#setting-up-for-development).
2. Perform the following QA steps:
    - Check that the pagination is working as expected in the instance list as well as the snapshot list.
    - Check that when changing the page size on a page that is not 1, the current page gets reset to 1.

[WD-2713]: https://warthogs.atlassian.net/browse/WD-2713?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
